### PR TITLE
[istio] adding default resources for init-containers

### PR DESCRIPTION
**Pull request by AndreyPavlovFlant**
## Description
Adding default resources for init containers. 

## Why do we need it, and what problem does it solve?
If `resourcequotas` are enabled, pods with containers that don't have limits or requests cannot start.
Fixed https://github.com/deckhouse/deckhouse/issues/8919

## What is the expected result?
We have better control over the use of resources in the cluster and also do not experience any problems when using `resourcequotas`.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: istio
type: chore
summary: For all new pods, the CPU limit will be set to 2 in the Istio sidecar container. If this value is too small for your cluster, you can set a higher value in the istio ModuleConfig. To apply the new limits to previously created pods, you will need to restart them manually.
impact: For all new Istio sidecar containers the CPU limit will be set to 2 (you can tune this). To apply the limit to existing pods, you will need to restart them manually.
impact_level: high
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've started the workflow for label 'e2e/run/aws'...

**Comment by AndreyPavlovFlant** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.27 9955852854 9955852854-2-con-1-27 /sys/deckhouse-oss/install:pr9036 ec2-user@3.124.168.214

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've started the workflow for label 'e2e/run/openstack'...

File: modules/110-istio/templates/control-plane/iop.yaml [hunk 1]
```diff
@@ -173,7 +182,7 @@ spec:
       proxy:
         image: {{ include "helm_lib_module_image" (list $ (printf "proxyv2%s" $imageSuffix )) }}
         clusterDomain: {{ $.Values.global.discovery.clusterDomain | quote }}
-  {{- if $.Values.istio.sidecar.resourcesManagement }}
+  {{- if and ($.Values.istio.sidecar.resourcesManagement) (eq $.Values.istio.sidecar.resourcesManagement.mode "Static") ($.Values.istio.sidecar.resourcesManagement.static)}}
```
Comment (apolovov) [inline modules/110-istio/templates/control-plane/iop.yaml:185]:
```suggestion
  {{- if and ($.Values.istio.sidecar.resourcesManagement) (eq $.Values.istio.sidecar.resourcesManagement.mode "Static") ($.Values.istio.sidecar.resourcesManagement.static) }}
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've started the workflow for label 'e2e/run/openstack'...

**Review by apolovov — APPROVED** [review]

**Review by z9r5 — APPROVED** [review]

**Pull request merged**